Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,334 @@
+            if len(img) > 100:
+                # Send back to Demisto inside function
+                return fileResult('topo.png', img, entryTypes['image'])
+            else:
+                entry['HumanReadable'] = 'No Valid Path Found'
+                entry['Contents'] = 'No Valid Path Found'
+                # Send back to Demisto inside function
+                return entry
+        except:
+            return_error('Error Running Query')
+
+    def path_finder_image_command():
+        """ Sample query: querystring = {'src':'10.80.80.0','dst':'172.16.200.80','service':'tcp:80','includeIncompletePaths':'true','displayBlockedStatus':'true'} """
+        querystring = {
+            'src': demisto.args()['source'],
+            'dst': demisto.args()['destination'],
+            'service': demisto.args().get('service', 'Any'),
+            'includeIncompletePaths': 'true',
+            'displayBlockedStatus': 'true'
+        }
+        e = path_finder_image(querystring)
+        demisto.results(e)
+
+    def device_name(devices, device_id):
+        return [e['name'] + ' ({} {})'.format(e['vendor'], e['model']) for e in devices if int(e['id']) == int(device_id)][0]
+
+    def object_lookup(querystring):
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        return_json = {'objects': []}
+
+        o = tos_request('/securetrack/api/network_objects/search', querystring)
+
+        # Validate result
+        try:
+            total = int(o['network_objects']['count'])
+        except KeyError:
+            total = 0
+
+        if total > 0:
+            device_json = tos_request('/securetrack/api/devices')['devices']['device']
+            objs = o['network_objects']['network_object']
+            if not isinstance(o['network_objects']['network_object'], list):
+                objs = [objs]
+            for obj in objs:
+                # display_name device_id
+                return_json['objects'].append({'object_name': obj['display_name'], 'device': device_name(device_json, obj['device_id']), 'comment': obj['comment']})
+        else:
+            entry['HumanReadable'] = 'No Results'
+            return entry
+
+        # Return to Demisto
+        entry['Contents'] = json.dumps(return_json)
+        entry['HumanReadable'] = tableToMarkdown('Object Lookup for {}'.format(querystring['exact_subnet']), return_json['objects'], ['object_name', 'device', 'comment'], underscoreToCamelCase, removeNull=True)
+        # Send back to Demisto inside function
+        return entry
+
+    def object_lookup_command():
+        """ Sample query: querystring = {'filter':'subnet','count':'50','exact_subnet':'1.1.1.1'} """
+        querystring = {
+            'filter': 'subnet',
+            'count': '50',
+            'exact_subnet': demisto.args()['ip']
```

Comment:
Need to improve the validation the input of the IP address
For example, if given a string which is not a valid IP address as input, a non indicative error is raised

Reply:
Added validation function before running command.